site/content/en/docs/tasks/dev/integrate_a_custom_job.md

```diff
@@ -6,63 +6,98 @@ description: >
+### Job Framework
 
-## Webhook
+In the `mycrd_controller.go` file in your folder, implement the `GenericJob` interface, and other optional [interfaces defined by the framework](https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/jobframework/interface.go).
 
-Create a webhook file in the dedicated directory for the integration.
+In the `mycrd_webhook.go` file in your folder, provide the webhook that invokes helper methods in from the jobframework to
+set the initial suspend status of created jobs and validate invariants.
 
-You can learn how to create webhook in this [page](https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html).
+Add testing files for both the controller and the webhook.  You can check the test files in the other subfolders of `./pkg/controller/jobs/`
+to learn how to implement them.
 
-The framework provides some validation functions for common labels and annotations.
+### Adjust build system
 
-Your webhook should have ability to suspend jobs.
+Add required dependencies to compile your code. For example, using `go get github.com/kubeflow/mpi-operator@0.4.0`.
 
+Update the [Makefile](https://github.com/kubernetes-sigs/kueue/blob/main/Makefile) for testing.
+   - Add commands which copy the CRD of your custom job to the Kueue project.
+   - Add your custom job operator CRD dependencies into `test-integration`.
 
-## Adjust build system
+## Building an External Integration
 
-1. Add required dependencies to compile the controller and webhook code. For example, using `go get github.com/kubeflow/mpi-operator@0.4.0`.
-2. Update [Makefile](https://github.com/kubernetes-sigs/kueue/blob/main/Makefile) for testing.
-   - Add commands which copy the crd of your custom job to the Kueue project.
-   - Add your custom job operator crd dependencies into `test-integration`.
+Here are completed external integrations you can learn from:
+   - [AppWrapper](https://github.com/project-codeflare/appwrapper)
 
-   For testing files, you can check the sample test files in [completed integrations](#completed-integrations) to learn how to implement them.
+### Registration
 
-## Completed integrations
-Here are completed integrations you can learn from:
-   - [BatchJob](https://github.com/kubernetes-sigs/kueue/tree/main/pkg/controller/jobs/job)
-   - [JobSet](https://github.com/kubernetes-sigs/kueue/tree/main/pkg/controller/jobs/jobset)
-   - [MPIJob](https://github.com/kubernetes-sigs/kueue/tree/main/pkg/controller/jobs/mpijob)
-   - [RayJob](https://github.com/kubernetes-sigs/kueue/tree/main/pkg/controller/jobs/rayjob)
\ No newline at end of file
+Add your framework's GroupVersionKind to `.integrations.externalFrameworks` in [controller_manager_config.yaml](https://github.com/kubernetes-sigs/kueue/blob/main/config/components/manager/controller_manager_config.yaml)
```

Comment:
Link to https://kueue.sigs.k8s.io/docs/installation/#install-a-custom-configured-released-version instead

Reply:
Link updated.  I also added externalFrameworks as a commented out stanza to the example config map.

I left the config map otherwise unchanged, but I was wondering if we really needed to have the disclaimer about fields that were only available in version 0.3 and later.